api/envoy/config/dynamic_parameters/v3/dynamic_parameters.proto

```diff
@@ -0,0 +1,19 @@
+syntax = "proto3";
+
+package envoy.config.dynamic_parameters.v3;
+
+import "envoy/service/discovery/v3/dynamic_parameter_constraints.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.config.dynamic_parameters.v3";
+option java_outer_classname = "DynamicParametersProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Dynamic Parameter Constraints Map Resource]
+
+message DynamicParameterConstraintsMap {
```

Comment:
Who uses this?

Reply:
Oh I see, this is the DCP subscription service..

Reply:
This will be used in the use-cases where we want the xDS server to dictate the dynamic parameter constraints sent by the client.  In the design doc, this is used in the "Sharding Endpoints" and "Selecting Cluster Based on ACL" use-cases.  (This is the mechanism we'd previously discussed in [xDS Context Parameter Negotiation](https://docs.google.com/document/d/1GO6HJ08wbMqLEjVGit5AxMiEAdHzZnT84f4A9Xg5Jp0/edit), although that doc hasn't yet been updated to reflect this new design.)

Reply:
Will this available through ADS?

Reply:
Yes, absolutely.  Any xDS resource type can be obtained via ADS.